xplique/concepts/craft_manager.py

```diff
@@ -0,0 +1,177 @@
+"""
+CRAFT MANAGER Module for Tensorflow/Pytorch
+"""
+
+from abc import ABC, abstractmethod
+
+import numpy as np
+
+from ..types import Callable, Optional
+from .craft import DisplayImportancesOrder
+
+class BaseCraftManager(ABC):
+    """
+    Base class implementing the CRAFT Concept Extraction Mechanism on multiple classes.
+    This manager creates one BaseCraft instance per class to explain.
+
+    It shall be subclassed in order to adapt to a specific framework
+    (Tensorflow or Pytorch)
+
+    Parameters
+    ----------
+    inputs
+        Input data of shape (n_samples, height, width, channels).
+        (x1, x2, ..., xn) in the paper.
+    labels
+        Labels of the inputs of shape (n_samples, class_id)
+    list_of_class_of_interest
+        A list of the classes id to explain. The manager will instanciate one
+        Craft object per element of this list.
+    """
+
+    @abstractmethod
+    def __init__(self,
+                input_to_latent_model : Callable,
+                latent_to_logit_model : Callable,
+                inputs : np.ndarray,
+                labels : np.ndarray,
+                list_of_class_of_interest : list = None):
+        self.input_to_latent_model = input_to_latent_model
+        self.latent_to_logit_model = latent_to_logit_model
+        self.inputs = inputs
+        self.labels = labels
+
+        if list_of_class_of_interest is None:
+            # take all the classes
+            list_of_class_of_interest = np.array(list(set(labels)))
+        self.list_of_class_of_interest = list_of_class_of_interest
+        self.craft_instances = None
+
+    @abstractmethod
+    def compute_predictions(self):
+        """
+        Compute the predictions for the current dataset, using the 2 models
+        input_to_latent_model and latent_to_logit_model chained.
+        To be implemented by decicated subclasses for Tensorflow or Pytorch.
+
+        Returns
+        -------
+        y_preds
+            the predictions
+        """
+        raise NotImplementedError
+
+    def fit(self, nb_samples_per_class: int = None):
```

Comment:
`nb_samples_per_class` can be `Optional[int]` here ;)

Reply:
done